3/3-Core-Structures.md

```diff
@@ -325,3 +387,1033 @@ Enumerates hash algorithms supported or known by the runtime for shared use.
+| COLORADO | 6 |  |
+| CT | 7 | State of Connecticut. |
+| CONNECTICUT | 7 |  |
+| DE | 8 | State of Delaware. |
+| Delaware | 8 |  |
+| DC | 9 | Washington, District of Columbia (DC). |
+| DISTRICT_OF_COLUMBIA | 9 |  |
+| FL | 10 | State of Florida. |
+| FLORIDA | 10 |  |
+| GA | 11 | State of Georgia. |
+| GEORGIA | 11 |  |
+| HI | 12 | State of Hawaii. |
+| HAWAII | 12 |  |
+| ID | 13 | State of Idaho. |
+| IDAHO | 13 |  |
+| IL | 14 | State of Illinois. |
+| ILLINOIS | 14 |  |
+| IN | 15 | State of Indiana. |
+| INDIANA | 15 |  |
+| IA | 16 | State of Iowa. |
+| IOWA | 16 |  |
+| KS | 17 | State of Kansas. |
+| KANSAS | 17 |  |
+| KY | 18 | State of Kentucky. |
+| KENTUCKY | 18 |  |
+| LA | 19 | State of Loisiana. |
+| LOISIANA | 19 |  |
+| ME | 20 | State of Maine. |
+| MAINE | 20 |  |
+| MD | 21 | State of Maryland. |
+| MARYLAND | 21 |  |
+| MA | 22 | State of Massachusetts. |
+| MASSACHUSETTS | 22 |  |
+| MI | 23 | State of Michigan. |
+| MICHIGAN | 23 |  |
+| MN | 24 | State of Minnesota. |
+| MINNESOTA | 24 |  |
+| MS | 25 | State of Mississippi. |
+| MISSISSIPPI | 25 |  |
+| MO | 26 | State of Missouri. |
+| MISSOURI | 26 |  |
+| MT | 27 | State of Montana. |
+| MONTANA | 27 |  |
+| NE | 28 | State of Nebraska. |
+| NEBRASKA | 28 |  |
+| NV | 29 | State of Nevada. |
+| NEVADA | 29 |  |
+| NH | 30 | State of New Hampshire. |
+| NEW_HAMPSHIRE | 30 |  |
+| NJ | 31 | State of New Jersey. |
+| NEW_JERSEY | 31 |  |
+| NM | 32 | State of New Mexico. |
+| NEW_MEXICO | 32 |  |
+| NY | 33 | State of New York. |
+| NEW_YORK | 33 |  |
+| NC | 34 | State of North Carolina. |
+| NORTH_CAROLINA | 34 |  |
+| ND | 35 | State of North Dakota. |
+| NORTH_DAKOTA | 35 |  |
+| OH | 36 | State of Ohio. |
+| OHIO | 36 |  |
+| OK | 37 | State of Oklahoma. |
+| OKLAHOMA | 37 |  |
+| OR | 38 | State of Oregon. |
+| OREGON | 38 |  |
+| PA | 39 | State of Pennsylvania. |
+| PENNSYLVANIA | 39 |  |
+| RI | 40 | State of Rhode Island. |
+| RHODE_ISLAND | 40 |  |
+| SC | 41 | State of South Carolina. |
+| SOUTH_CAROLINA | 41 |  |
+| SD | 42 | State of South Dakota. |
+| SOUTH_DAKOTA | 42 |  |
+| TN | 43 | State of Tennessee. |
+| TENNESSEE | 43 |  |
+| TX | 44 | State of Texas. |
+| TEXAS | 44 |  |
+| UT | 45 | State of Utah. |
+| UTAH | 45 |  |
+| VT | 46 | State of Vermont. |
+| VERMONT | 46 |  |
+| VA | 47 | State of Virginia. |
+| VIRGINIA | 47 |  |
+| WA | 48 | State of Washington. |
+| WASHINGTON | 48 |  |
+| WV | 49 | State of West Virginia. |
+| WEST_VIRGINIA | 49 |  |
+| WI | 50 | State of Wisconsin. |
+| WISCONSIN | 50 |  |
+| WYOMING | 51 | State of Wyoming. |
+| WY | 51 |  |
+
+## `opencannabis.device`
+
+Specifies a variety of devices, device flags, device types, and so on.
+
+{% nomnoml %}
+
+#fill: #d5e7ee; #8ebff2
+[Device
+ | uuid: string
+ | type: DeviceType
+ | flags: DeviceFlags
+ | key: DeviceCredentials]
+
+[DeviceCredentials
+ | public_key: bytes
+ | private_key: bytes
+ | sha256: string
+ | identity: string
+ | authorities: bytes: repeated]
```

Comment:
we should probably express repeated properties as `repeated bytes`, not `bytes: repeated`